avm/ptn/azd/ml-project/main.bicep

```diff
@@ -0,0 +1,171 @@
+@metadata({
+  example: '''
+  {
+      "key1": "value1"
+      "key2": "value2"
+  }
+  '''
+})
+param tags object?
+
+@description('Optional. Specifies the SKU, also referred as \'edition\' of the Azure Machine Learning workspace.')
+@allowed([
+  'Free'
+  'Basic'
+  'Standard'
+  'Premium'
+])
+param projectSku string = 'Basic'
+
+@description('Optional. The type of Azure Machine Learning workspace to create.')
+@allowed([
+  'Default'
+  'Project'
+  'Hub'
+  'FeatureStore'
+])
+param projectKind string = 'Project'
+
+@description('Optional. The flag to signal HBI data in the workspace and reduce diagnostic data collected by the service.')
+param hbiWorkspace bool = false
+
+@description('Optional. Whether or not public network access is allowed for this machine learning resource. For security reasons it should be disabled.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = 'Enabled'
+
+@description('Required. The resource ID of the AI Studio Hub Resource where this project should be created.')
+param hubResourceId string
+
+@description('Required. The name of the key vault.')
+param keyVaultName string
+
+@description('Optional. The managed identity definition for the machine learning resource. At least one identity type is required.')
+param projectManagedIdentities managedIdentitiesType = {
+  systemAssigned: true
+}
+
+@description('Required. The name of the user assigned identity.')
+param userAssignedName string
+
+@description('Optional. The role to assign. You can provide either the display name of the role definition, the role definition GUID, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'. Default roles: AzureML Data Scientist, Azure Machine Learning Workspace Connection Secrets Reader.')
+param roleDefinitionIdOrName array = [
+  'f6c7c914-8db3-469d-8ca1-694a8f32e121' // AzureML Data Scientist
+  'ea01e6af-a1c1-4350-9563-ad00f8c72ec5' // Azure Machine Learning Workspace Connection Secrets Reader
+]
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.azd-mlproject.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+module project 'br/public:avm/res/machine-learning-services/workspace:0.7.0' = {
+  name: '${uniqueString(deployment().name, location)}-project'
+  params: {
```

Comment:
Misses passthru of `enableTelemetry` parameter

Reply:
@Menghua1, please note, you did not add a passthru (`enableTelemetry: enableTelemetry`) but instead added the telemetry as a condition for the module's deployment (which you certainly don't want) 😉 

Please fix this for all module references. Also, I'd recommend to not resolve comments yourself, but simply comment them once you tackled them. Otherwise re-review are quite painful.